lxd/api_internal.go

```diff
@@ -172,16 +172,17 @@ func internalCreateWarning(d *Daemon, r *http.Request) response.Response {
 		return response.BadRequest(err)
 	}
 
-	req.EntityTypeCode, _ = reqRaw.GetInt("entity_type_code")
+	entityTypeStr, _ := reqRaw.GetString("entity_type")
 	req.EntityID, _ = reqRaw.GetInt("entity_id")
 
-	// Check if the entity exists, and fail if it doesn't.
-	_, ok := cluster.EntityNames[req.EntityTypeCode]
-	if req.EntityTypeCode != -1 && !ok {
-		return response.SmartError(fmt.Errorf("Invalid entity type"))
+	// If entity type is set, check it is valid and fail if it isn't.
+	req.EntityType = entity.Type(entityTypeStr)
+	err = req.EntityType.Validate(true)
```

Comment:
Can we remove the argument to Validate here, and instead test of entityTypeStr is empty and not call Validate in that case?